__tests__/services/users.service.spec.ts

```
@@ -30,3 +39,84 @@ describe('compare password', () => {
     expect(result).toBeFalsy();
   });
 });
+
+describe('verifying an account', () => {
+  beforeEach(async () => {
+    user = await factory(User)().create();
+  });
+
+  beforeAll(async () => {
+    usersService = Container.get(UsersService);
+    userRepository = getRepository<User>(User);
+  });
+
+  let user: User;
+  let userRepository: Repository<User>;
+
+  it('checks the hash is invalid', async () => {
```

(3) I think a better name for this test case could be `it should throw an Error if the hash is invalid. Again, you should describe what the test case is testing and not what the code does.

In the future, if someone modifies your original function this test will fail (that's the main purpose of the unit tests basically) so you should describe what is testing and not what is the code going to do because you will never know if changes.